### PR TITLE
Improve statehash analyzer

### PR DESCRIPTION
**Pull request by nickeskov**

### CODE REVIEW

File: pkg/analysis/analyzer.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ func (a *Analyzer) analyze(alerts chan<- entities.Alert, pollingResult *entities
 		go func(f func(in chan<- entities.Alert) error) {
 			defer wg.Done()
 			if err := f(criteriaOut); err != nil {
+				// TODO: send internal analyzer alert
```
Comment (nickeskov) [inline pkg/analysis/analyzer.go:99]: This `TODO` was fixed in #108

File: pkg/entities/alerts.go [hunk 2]
```diff
@@ -356,3 +356,5 @@ func (a *AlertFixed) Type() AlertType {
 func (a *AlertFixed) Level() string {
 	return InfoLevel
 }
+
+// TODO: create internal error alert for analyzer
```
Comment (nickeskov) [inline pkg/entities/alerts.go:360]: This `TODO` was fixed in #108

File: pkg/analysis/criteria/state_hash.go [hunk 3]
```diff
@@ -21,32 +27,52 @@ type StateHashCriterion struct {
...
+		for _, statement := range nodeStatements {
+			var statementAtBucketHeight entities.NodeStatement
+			if statement.Height == bucketHeight {
+				statementAtBucketHeight = statement
+			} else {
+				var err error
+				statementAtBucketHeight, err = c.es.GetStatementAtHeight(statement.Node, bucketHeight)
+				if err != nil {
+					if errors.Is(err, events.NoFullStatementError) {
+						// TODO: should we ignore statement in such case?
```
Comment (esuwu) [inline pkg/analysis/criteria/state_hash.go:50]: The PR with zap logger has been merged, you can now add a log here

**Review by esuwu — CHANGES_REQUESTED** [review]

File: pkg/analysis/criteria/state_hash.go (hunk 3 above)
Comment (nickeskov) [reply pkg/analysis/criteria/state_hash.go]: Fixed

**Review by alexeykiselev — APPROVED** [review]

**Pull request merged**